src/RuleFilterParser/Api/ApiRequest.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace RuleFilterParser;
+
+public class ApiRequest : IApiRequest
```

Comment:
It looks like we can combine ApiRequest and ApiQueryBuilder

Reply:
Yes, that could be done.

But I think I made here a mistake in the design:

As I understood you, you wanted to have a separate request model class per request so that it is possible to make separate Swagger documentation per API request. That was the reason I implemented `IApiRequest`.
The different request models (e.g. [`GetTargetsRequest`](https://github.com/bcc-code/bcc-contributions-api/blob/main/src/Contributions.Api/Models/Requests/GetTargetsRequest.cs)) would then implement the `IApiRequest` interface.

What I would need to add is a method in interface `IApiClient` which creates the `IApiRequest` for the specific request. We could implement a method `IApiRequest CreateApiRequest<T>(IApiRequest)` for that. Alternatively I could check to find out if it is possible to tell Swagger that for the documentation for an `IApiRequest` paramter another class should be used. I don't know if that is possible.

Reply:
class `ApiQueryBuilder` is now removed